app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -258,6 +258,7 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
 
     override fun onBackPressed(): Boolean {
         invokePendingDeletion()
+        sharedViewModel.selectedFolder = null
```

Comment:
Looks like the logic could be inside of [DefaultBookmarkController](https://github.com/mozilla-mobile/fenix/blob/17d9034013b51cbbcb650ce8ff1334b20102ebdd/app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt#L52), I see similar functions there like [handleBookmarkChanged](https://github.com/mozilla-mobile/fenix/blob/17d9034013b51cbbcb650ce8ff1334b20102ebdd/app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt#L70), maybe we need something like `handleClearSelectedFolder` and then you can test it on [DefaultRecentBookmarksControllerTest](https://github.com/mozilla-mobile/fenix/blob/master/app/src/test/java/org/mozilla/fenix/home/recentbookmarks/DefaultRecentBookmarksControllerTest.kt), what do you think? :)

Reply:
Sure, I'll look into it.  Thanks,

Reply:
@Amejia481 It's a good idea but while trying to implement it I noticed that multiple places does the same assignment: `BookmarkFragment`, `AddBookmarkFolderFragment` and `EditBookmarkFragment`.  Since in each case `sharedViewModel` is owned on the Fragment level, I think this call should stay in `BookmarkFragment`.

Reply:
No problem!